Sources/Commands/SwiftTool.swift

```diff
@@ -618,13 +620,14 @@ public class SwiftTool {
                 createMultipleTestProducts: createMultipleTestProducts,
                 createREPLProduct: createREPLProduct,
                 forceResolvedVersions: options.resolver.forceResolvedVersions,
+                skipResolve: skipResolve,
                 testEntryPointPath: testEntryPointPath,
                 observabilityScope: self.observabilityScope
             )
 
             // Throw if there were errors when loading the graph.
             // The actual errors will be printed before exiting.
-            guard !self.observabilityScope.errorsReported else {
+            if self.observabilityScope.errorsReported && exitOnError {
```

Comment:
why is this change required?

Reply:
In case package resolve is skipped with the new option, it can result in not finding a dependent product, which will then throw an error and then exits. This will ensure an error msg in dumpDependencies func is displayed.